specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+        String Title;
+
+        // Gets/Sets the Value for the numerical Progress percentile: a number between 0 and 1. Binds to {progressValue} in progress xml tag.
+        Double Value;
+
+        // Gets/Sets the Value for the Progress String. Binds to {progressValueString} in progress xml tag
+        String ValueStringOverride;
+
+        // Gets/Sets the Value for the Status. Binds to {progressStatus} in progress xml tag
+        String Status;
+    }
+
+    // The Notification User Setting or Notification Group Policy Setting
+    enum AppNotificationSetting
+    {
+        Enabled, // Notification is not blocked by settings or group policy
+        DisabledForApplication, // Notification is blocked by a user defined App Setting
+        DisabledForUser, // Notification is blocked by a user defined Global Setting
+        DisabledByGroupPolicy, // Notification is blocked by Group Policy
+        DisabledByManifest, // Notification is blocked by a setting in the manifest. Only for packaged applications.
+    };
+
+    // The Result for a Notification Progress related operation
+    enum AppNotificationProgressResult
+    {
+        Succeeded, // The progress operation succeeded
+        AppNotificationNotFound, // The progress operation failed to find a Notification to process updates
+    };
+
+    // The Priority of the Notification UI associated with it's popup in the Action Centre
+    enum AppNotificationPriority
+    {
+        Default, // The notification should have default behavior in terms of delivery and display priority during connected standby mode.
+        High, // The notification should be treated as high priority. For desktop PCs, this means during connected standby mode the incoming notification can turn on the screen for Surface-like devices if it doesn't have a closed lid detected.
+    };
+
+    runtimeclass AppNotification
+    {
+        AppNotification(Windows.Data.Xml.Dom.XmlDocument payload);
+
+        // Unique identifier used to replace a notification within a group.
+        String Tag;
+
+        // Unique identifier for a Notification group in the app
+        String Group;
+
+        // A unique identifier for the Notification generated by the platform.
+        UInt32 Id { get; };
+
+        // The notification Xml Payload
+        Windows.Data.Xml.Dom.XmlDocument Payload{ get; };
+
+        // Gets or sets additional information about the Notification progress.
+        AppNotificationProgressData Progress;
+
+        // Gets or sets the time after which a Notification should not be displayed.
+        Windows.Foundation.DateTime Expiration;
+
+        // Indicates whether the Notification will remain in the Action Center after a reboot.
+        Boolean ExpiresOnReboot;
+
+        // Gets or sets the priority for a Notification.
+        // Hints on how and at what urgency level a notification should be presented to the user (whether to wake up the screen, etc).
```

Comment:
Priority or urgency? They are different concepts; we should use the appropriate terms (in API and in docs).

Reply:
Priority is consistent with Windows SDK and is the consistent naming convention.